### ISSUE TITLE
Zce LLVM (for prototype)

### ISSUE THREAD

**Issue by jjscheel** [riscv-admin/dev-partners#4]
### Technical Group

Code Size Reduction TG

### ratification-pkg

Code Size

### Technical Liaison

Tariq Kurd

### Task Category

Compilers/Toolchain

### Task Sub Category

- [ ] gcc
- [ ] binutils
- [ ] gdb
- [ ] intrinsics
- [ ] Java
- [ ] KVM
- [ ] ld
- [X] llvm
- [ ] Linux kernel
- [ ] QEMU
- [ ] Spike

### Ratification Target

3Q2023

### Statement of Work (SOW)

[SOW link](https://docs.google.com/document/d/1k0xdoMMUpNZ6f4vGy0B0_66RYecToKEoIoSc9P59chA/)

**SOW Signoffs:** (delete those not needed)
  - Task group liaison sign-off date:
  - Development partner sign-off date:
  - Toolchains & Runtimes sign-off date (if gcc, LLVM, binutils, gdb work):

### Waiver

- [ ] Freeze
- [ ] Ratification

**Comment by jjscheel** [comment]
Patches remain in review per 3/28 meeting.  No action needed.

**Comment by jjscheel** [comment]
@lazyparser, would you kindly help me get this assigned to Chunyu?

**Comment by ChunyuLiao** [comment]
GitHub ID: ChunyuLiao, You can assign it to me, sorry for the late reply.

**Comment by jjscheel** [comment]
Thanks, @ChunyuLiao.  Would you kindly also make sure that you've added it to your RISC-V Profile in the GitHub id field.  Details are here: [link](https://lists.riscv.org/g/dev-partners/topic/risc_v_profile_update/97658653)

Also, if you're not a member of the [DevPartners community](https://lists.riscv.org/g/dev-partners), please join.

**Comment by jjscheel** [comment]
@ChunyuLiao, nevermind.  I see that you've already done this.  THANKS!!!

**Comment by jjscheel** [comment]
Thanks, @ChunyuLiao!  I've bumped the projected completion date to next meeting, April 25.

**Comment by jjscheel** [comment]
Based on last week's discussion that reviews are still pending, I've bumped the Completion Projection date to next meeting (May 9) to match Next Status date.

**Comment by ChunyuLiao** [comment]
[RISCV] Add MC support of RISCV zcmt Extension  https://reviews.llvm.org/D134600  has been merged

**Comment by jjscheel** [comment]
Thanks, @ChunyuLiao!  Great news!!!  1 down 3 to go.

**Comment by jjscheel** [comment]
Removed from Agenda for this week.

**Comment by Xinlong-Wu** [comment]
[RISCV] Add MC support of RISCV zcmp Extension https://reviews.llvm.org/D132819 has been merged

**Comment by jjscheel** [comment]
Way to go!  Just 2 more!!!

**Comment by jjscheel** [comment]
Done.  Thank you, @ChunyuLiao!

**Comment by ChunyuLiao** [comment]
All Zcmp patchs have been merged.

**Comment by jjscheel** [comment]
Marking as done.  Thanks, @ChunyuLiao!  Nice work!!!

**Comment by ChunyuLiao** [comment]
Sorry Jeff.  There is only one zcmt not merged and it is the 6th patch https://reviews.llvm.org/D134600.

**Comment by jjscheel** [comment]
Ok, thanks, @ChunyuLiao.  I've set this back to "Reviewing".  We are VERY close to being able to celebrate success!!!

**Comment by jjscheel** [comment]
@ChunyuLiao, would you kindly provide status here.

**Comment by ChunyuLiao** [comment]
No new updates lately. waiting the reviewer of lld help review the lld patch https://reviews.llvm.org/D134600 again.

**Comment by jjscheel** [comment]
Thanks, @ChunyuLiao.

**Comment by jjscheel** [comment]
@ChunyuLiao, have you received any new feedback on the lld patch?

**Comment by ChunyuLiao** [comment]
> @ChunyuLiao, have you received any new feedback on the lld patch?

Yes, jrtc27 helped review the patch again and the patch has been updated.

**Comment by ChunyuLiao** [comment]
No updates for the last two weeks.

**Comment by jjscheel** [comment]
Thanks, @ChunyuLiao!

**Comment by ChunyuLiao** [comment]
No update for the last two weeks. thanks.

**Comment by jjscheel** [comment]
Thanks, @ChunyuLiao.  Have you received any feedback on your latest patch?

**Comment by ChunyuLiao** [comment]
> Thanks, @ChunyuLiao. Have you received any feedback on your latest patch?

No feedback. I'm guessing there won't be any feedback for the next month.

**Comment by ChunyuLiao** [comment]
> Your last update on the patch was 7th September. If you want people to review it you should ping the patch (adhering to https://llvm.org/docs/CodeReview.html#code-reviews-speed-and-reciprocity) and possibly them specifically, otherwise it drops off their radar.
Thanks, we're looking forward to your suggestions!

**Comment by ChunyuLiao** [comment]
Waiting for re-review.
 We've done some testing and the results have been added to the comments, hope it helps.

**Comment by Xinlong-Wu** [comment]
Still in reviewing
Since the [reviews.llvm.org](https://reviews.llvm.org/D99750) has been turn to static, the patch is hard to be access and review patch.
So we are trying to move it to github PR.

**Comment by ChunyuLiao** [comment]
new pr: https://github.com/llvm/llvm-project/pull/77884

**Comment by jjscheel** [comment]
Thanks, @ChunyuLiao.  Added to the list in header.

**Comment by ChunyuLiao** [comment]
No new updates for the last 3 weeks.

**Comment by ChunyuLiao** [comment]
No updates for the last 2 weeks.

**Comment by ChunyuLiao** [comment]
Do more discussions and updated zcmt last week.

**Comment by ChunyuLiao** [comment]
no update.

**Comment by jjscheel** [comment]
@Xinlong-Wu, @ChunyuLiao, anything we can do to encourage review of the PR?
@cmuellner, are you aware of this PR and any problems here?

**Comment by Xinlong-Wu** [comment]
no update for pervious weeks.

currently， the reviewers are concerned about the efficiency(how much reduction we can get by current implementation) of zcmt can do. 

It related to the linking time, so the reviewers would be very strict

**Comment by Xinlong-Wu** [comment]
no update

**Comment by Xinlong-Wu** [comment]
no more update

**Comment by jjscheel** [comment]
@cmuellner, it seems that we're struggling to get a review on this patch.  what suggestions do you have to help @ChunyuLiao get feedback?

**Comment by Xinlong-Wu** [comment]
no update

**Comment by jjscheel** [comment]
@cmuellner, any advice on how to proceed here?

**Comment by Xinlong-Wu** [comment]
nothing more

**Comment by Xinlong-Wu** [comment]
no update

**Comment by jjscheel** [comment]
@ptomsich, @cmuellner, @kito-cheng, any suggestions to help with this PR? https://github.com/llvm/llvm-project/pull/77884

**Comment by cmuellner** [comment]
This has likely fallen through the cracks. It happens sometimes, and the best way to get it up on the table is to be persistent and write a "ping" message (or ask for the next steps) in the PR thread if the waiting time exceeds two weeks.

I've asked for the next steps in the PR thread.

There is also an LLVM/RISC-V community call every second week, during which this could be discussed with maintainers and leading developers.

**Comment by cmuellner** [comment]
And sorry for failing to see the earlier reach-out messages to me!

**Comment by jjscheel** [comment]
Thanks, @cmuellner!!!

**Comment by Xinlong-Wu** [comment]
No more updates, plan to recompile some software in the following four weeks to evaluate the efficiency of the current implementation

**Comment by jjscheel** [comment]
@Xinlong-Wu, please consider @cmuellner's suggestions about how to get PR consideration.

**Comment by kito-cheng** [comment]
@Xinlong-Wu
Chatted with MaskRay in person in the LLVM dev meeting (which same week as RISC-V summit), he is OK to moving forward if you rebase the PR, his most concern was no vendor really implement that in real product, however Synopsys folks and me has express we did ship that extension within our product, so he is OK now.

**Comment by RobinKastberg** [comment]
It seemed work in the original Zcmt PR has stopped.
FYI I have continued work on this PR here: 
https://github.com/llvm/llvm-project/pull/163142

**Comment by jjscheel** [comment]
Still not complete.  No need to continue tracking here.

Closing.
